internal/datastore/common/relationships.go

```diff
@@ -0,0 +1,164 @@
+package common
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/authzed/spicedb/pkg/datastore"
+	corev1 "github.com/authzed/spicedb/pkg/proto/core/v1"
+	"github.com/authzed/spicedb/pkg/tuple"
+)
+
+const errUnableToQueryRels = "unable to query relationships: %w"
+
+// StaticValueOrAddColumnForSelect adds a column to the list of columns to select if the value
+// is not static, otherwise it sets the value to the static value.
+func StaticValueOrAddColumnForSelect(colsToSelect []any, queryInfo QueryInfo, colName string, field *string) []any {
```

Comment:
They are actually string reference, and it returns an `[]any` because that's what the `rows.Scan` requires.

This isn't adding the static value to the slice, its adding the reference to the string value for collecting the non-static value